devcluster/config.py

```diff
@@ -180,13 +192,97 @@ def build_stage(
 
         custom_config = CustomDockerConfig(
             {
-                "name": "db",
+                "name": self.name,
                 "container_name": self.container_name,
                 "kill_signal": "TERM",
                 "run_args": run_args,
-                "post": [
-                    {"logcheck": {"regex": "listening on IP"}},
-                ],
+                "post": self.post,
+            }
+        )
+
+        return dc.DockerProcess(custom_config, poll, logger, state_machine)
+
+
+class ElasticConfig(StageConfig):
+    """ElasticConfig is a canned stage that runs Elastic in docker"""
+
+    def __init__(self, config: typing.Any) -> None:
+        allowed = {
+            "name",
+            "api_port",
+            "elastic_port",
+            "environment",
+            "data_dir",
+            "container_name",
+            "image_name",
+            "cmdline",
+            "post",
+        }
+        required = set()  # type: typing.Set[str]
+        check_keys(allowed, required, config, type(self).__name__)
+
+        self.api_port = int(config.get("api_port", 9200))
+        self.elastic_port = int(config.get("elastic_port", 9300))
+
+        self.environment = check_dict_of_strings(
+            config.get(
+                "environment",
+                {
+                    "discovery.type": "single-node",
+                    "cluster.routing.allocation.disk.threshold_enabled": "false",
+                    "logger.level": "WARN",
+                },
+            ),
+            "ElasticConfig.environment must be a dict of strings to strings",
+        )
+        self.data_dir = read_path(config.get("data_dir"))
+        self.container_name = str(config.get("container_name", "determined_elastic"))
+        self.name = str(config.get("name", "elastic"))
+        self.image_name = str(
+            config.get(
+                "image_name", "docker.elastic.co/elasticsearch/elasticsearch:7.9.3"
```

Comment:
Why all this docker.elastic.co?  Why not just `elasticsearch:VERSION`?  And why does the [elasticsearch docker page](https://hub.docker.com/_/elasticsearch/) say that `7.9.3` isn't a supported image tag?

Reply:
>Why all this docker.elastic.co? Why not just elasticsearch:VERSION?

I was just porting this from `make -C tools`.

>And why does the elasticsearch docker page say that 7.9.3 isn't a supported image tag?

here it is, no? https://hub.docker.com/_/elasticsearch/?tab=tags&page=1&ordering=last_updated&name=7.9.3

Reply:
> I was just porting this from make -C tools.

@stoksc ?

> here it is, no?

Yeah you can download 7.9.3, but that link clearly says the supported elasticsearch tags are  7.14.0 and 6.8.18.  (@stoksc should I upgrade this tag to 7.14.0?)

Reply:
Looks like 7.14.0 seems to work fine.